lib/modules/platform/local/scm.ts

```diff
@@ -0,0 +1,45 @@
+import { glob } from 'glob';
+import type { CommitFilesConfig, CommitSha } from '../../../util/git/types';
+import type { PlatformScm } from '../types';
+
+let fileList: string[] | undefined;
+export class LocalFs implements PlatformScm {
+  fileList: string[] | undefined;
+
+  isBranchBehindBase(branchName: string, baseBranch: string): Promise<boolean> {
+    return Promise.resolve(false);
+  }
+  isBranchModified(branchName: string): Promise<boolean> {
+    return Promise.resolve(false);
+  }
+  isBranchConflicted(baseBranch: string, branch: string): Promise<boolean> {
+    return Promise.resolve(false);
+  }
+  branchExists(branchName: string): Promise<boolean> {
+    return Promise.resolve(true);
+  }
+  getBranchCommit(branchName: string): Promise<string | null> {
+    return Promise.resolve(null);
+  }
+  deleteBranch(branchName: string): Promise<void> {
+    return Promise.resolve();
+  }
+  commitAndPush(commitConfig: CommitFilesConfig): Promise<string | null> {
+    return Promise.resolve(null);
+  }
+
+  async getFileList(): Promise<string[]> {
+    fileList ??= await glob('**', {
+      dot: true,
+      nodir: true,
+      ignore: {
+        childrenIgnored: (p) => p.isNamed('.git') || p.isNamed('node_modules'),
```

Comment:
I needed to add these otherwise getFileList() took an eternity.

- probably should remove the hardcoded node_modules
- ideally it would automatically ignore everything in `.gitignore`
- `globby` does the above automatically, but it's on v13 with ESM and v11 was the last release to support non-ESM

Reply:
why not call git?